src/useSet.ts

```diff
@@ -13,14 +14,13 @@ export interface Actions<K> extends StableActions<K> {
 const useSet = <K>(initialSet = new Set<K>()): [Set<K>, Actions<K>] => {
   const [set, setSet] = useState(initialSet);
 
-  const stableActions = useMemo<StableActions<K>>(
-    () => ({
-      add: item => setSet(prevSet => new Set([...Array.from(prevSet), item])),
-      remove: item => setSet(prevSet => new Set(Array.from(prevSet).filter(i => i !== item))),
-      reset: () => setSet(initialSet),
-    }),
-    [setSet]
-  );
+  const stableActions = useMemo<StableActions<K>>(() => {
+    const add = (item: K) => setSet(prevSet => new Set([...Array.from(prevSet), item]));
+    const remove = (item: K) => setSet(prevSet => new Set(Array.from(prevSet).filter(i => i !== item)));
+    const toggle = (item: K) => (set.has(item) ? remove : add)(item);
```

Comment:
The toggle does not add `hello` back to the set in the story for me. I think this has to do with the `set` used in this function refers to a previous value. Using the following toggle function results in the desired toggle behaviour

```ts
const toggle = (item: K) =>
  setSet(prevSet => {
    if (prevSet.has(item)) {
      return new Set(Array.from(prevSet).filter(i => i !== item));
    } else {
      return new Set([...Array.from(prevSet), item]);
    }
  });
```

The test passes for both versions so I'm not sure if the test is testing the right things?

Reply:
Indeed! That seemed so straightforward that I didn't test it manually enough, in the end closures are always trickier than expected 😄 

Fixed it and updated the test to catch this kind of behaviour.